core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -767,7 +767,7 @@ class ControllerApis(val requestChannel: RequestChannel,
           setErrorCode(TOPIC_AUTHORIZATION_FAILED.code))
       }
     }
-    controller.createPartitions(context, topics).thenApply { results =>
+    controller.createPartitions(context, topics, request.validateOnly()).thenApply { results =>
```

Comment:
Good catch, it seems we don't have a test case for this problem, maybe we can change `PlaintextAdminIntegrationTest` to support KRaft mode.